app/controllers/sentences_controller.php

```diff
@@ -1033,14 +1054,24 @@ public function several_random_sentences()
 
         $randomIds = $this->Sentence->getSeveralRandomIds($lang, $number);
 
-        $this->Session->write('random_lang_selected', $lang);
+        if(is_bool($randomIds) && !$randomIds) {
+            $searchDisabled = !Configure::read('Search.enabled');
+            $this->set('searchProblem', true);
+            if ($searchDisabled) {
```

Comment:
Performa​n​ce is not really an issue ​in this case but I see the problem.

​There's an ambiguity in `getSeveralRandomIds()`. The function returns the same value whether the search is disabled or whether there's an error.

One possibility would be to return a ​certain value in case of disabled search, and another in case of search error.

But I would rather suggest to remove the `Configure::read('Search.enabled');` from `getSeveralRandomIds()` because it shouldn't be the job of the model to deal with the config.
Then in the controllers, wherever we retrieve random sentences, you would first check ​in the config ​if the search is disabled, and only when the search is not disabled, you would try to retrieve random sentences.